src/NJsonSchema/Generation/DefaultReflectionService.cs

```diff
@@ -202,7 +202,8 @@ public virtual JsonTypeDescription GetDescription(ContextualType contextualType,
                 return JsonTypeDescription.CreateForDictionary(contextualType, JsonObjectType.Object, isNullable);
             }
 
-            if (IsArrayType(contextualType) && contract is JsonArrayContract)
+            // TODO: Don't trust JsonArrayContract when contextualType is IAsyncEnumerable<T> until it is fixed
+            if (IsIAsyncEnumerableType(contextualType) || IsArrayType(contextualType) && contract is JsonArrayContract)
```

Comment:
`if (IsIAsyncEnumerableType(contextualType) || (IsArrayType(contextualType) && contract is JsonArrayContract))`
for safety

Reply:
Fixed